### ISSUE TITLE
Disable preview of csproj files / select default mode

### ISSUE THREAD

**Issue by puschie286** [nrwl/nx-console#2273]
The main purpose of opening a csproj file is to manual edit something - the preview mode is not useful most of the time, its just adds an additional click to change the view mode to "Editor". Please make the default mode "Editor" or add an configurable option for this.

## Current Behavior

csproj files are always open in "Editor and Preview" mode.

## Expected Behavior

csproj files should open in "Editor" mode.

## Steps to Reproduce

1. Open a csproj file

### Environment

JetBrains Rider 2024.2.5

**Comment by puschie286** [comment]
The last option sounds like the most intuitive behavior to me.  In the edge case of having multiple csproj files open and changing a single of them to another mode, it might feels like an bug.

So a complete solution could be an combination of the last two points. Therefore the option 'project file view mode'  should consist of 4 entries: "Editor", "Editor and Preview", "Preview" and "Last used mode". The "Last used mode" should probably be the default

This should would cover all possible cases of usage i can think of

**Comment by MaxKless** [comment]
#i would save the last opened editor state by file path so you can have individual preferences per file, even different ones for multiple `.csproj` files. That way no extra setting would be required, it can all be saved in the background.

**Comment by puschie286** [comment]
Well, how would i set the default mode ? (or change it later on).
I think that is the main use case

**Comment by MaxKless** [comment]
Yeah that's a good point, adding a default mode setting is a good idea, let's do it! 
Would you be interested in contributing here? It shouldn't be too difficult and I can help point you to the right places :)

**Comment by puschie286** [comment]
Yeah, i will try it - but it will probably take me some days as my schedule is pretty tight.

Thanks for the offer, i will come back to you if I have any problems :)

**Comment by MaxKless** [comment]
no worries, I look forward to what you cook up :) 
Have a look at the `apps/intellij/src/main/kotlin/dev/nx/console/settings` folder for previous art on building a setting. 
You'll need to look into the Kotlin UI DSL to build the setting form fields. 
You'll probably need to use the `PropertiesComponent` class to save the last used state information in a local file in the `.idea` folder. 
You can read the setting & last opened state in `apps/intellij/src/main/kotlin/dev/nx/console/project_details/ProjectDetailsEditorWithPreview.kt ` where the initial layout is set. 

Hope that helps :)

**Comment by pliersch** [comment]
Same for package.json please.

**Comment by MaxKless** [comment]
hey folks, I changed the default for the PDV to be closed. So I guess this issue can be closed. Do you think it's important to have it be auto-opened for some files?
